### ISSUE TITLE
Functions accepting type[T] cannot be typed in a way so that they accept annotated types

### ISSUE THREAD

**Issue by Wizzerinus** [microsoft/pyright#10395]
**Describe the bug**
I am not sure if this is a bug, because the documentation on Annotated is *scarce*, and already sensing the "Pyright is working as intended" 😄. However, I could not find a good way around this.

When `Annotated[SomeType, ...]` is passed to a function expecting `type[A]`, an error is triggered. Even if the function is changed to accept `type[Annotated[...]]` (I think? I'm not even sure what the syntax should look like here), it still causes an error, as Annotated is erased on the definition site but not on the call site. 

**Code or Screenshots**
```python
from typing import Annotated, TypeVar

T = TypeVar("T")
CoolInt = Annotated[int, "a"]

def getValue(t: type[T]) -> T | None:
    raise RuntimeError("not implemented")
def getValue2(t: type[Annotated[T, ...]]) -> T | None:
    raise RuntimeError("not implemented")


print(getValue(int))   # ok (as expected)
# Both of these calls error out
print(getValue(CoolInt))
print(getValue2(CoolInt))
```

**VS Code extension or command-line**
CLI, Pyright 1.1.399

**Comment by LeeeeT** [comment]
`type[]` is only meant for classes, not arbitrary type annotations (like `Annotated[]`). You could use PEP747's [`TypeForm`](https://peps.python.org/pep-0747/#why-not-type-c) once the PEP is implemented.

For now, I think the best you can do is to make an overload like this:

```python
@overload
def getValue(t: type[T]) -> T | None: ...

@overload
def getValue(t: Any) -> Any | None: ...
```

**Comment by erictraut** [comment]
Yes, pyright is working correctly here and in conformance with the typing spec. If you want to read more about how `type[]` works, refer to [this section](https://typing.python.org/en/latest/spec/special-types.html#type) of the typing spec. For information about `Annotated`, refer to [this section](https://typing.python.org/en/latest/spec/qualifiers.html#annotated).

If you have questions about the type system, the [Python typing forum](https://discuss.python.org/c/typing/32) is a good place to post. Or if you have a question about pyright specifically, feel free to post to the [discussion forum](https://github.com/microsoft/pyright/discussions) in the pyright issue tracker.
